platform-emf/src/test/java/com/softicar/platform/emf/attribute/field/enums/table/row/EmfEnumTableRowInputEngineTest.java

```diff
@@ -0,0 +1,80 @@
+package com.softicar.platform.emf.attribute.field.enums.table.row;
+
+import com.softicar.platform.common.core.i18n.LanguageEnum;
+import com.softicar.platform.common.core.locale.CurrentLocale;
+import com.softicar.platform.common.core.locale.Locale;
+import com.softicar.platform.db.runtime.test.AbstractDbTest;
+import java.util.Collection;
+import java.util.stream.Collectors;
+import org.junit.Test;
+
+public class EmfEnumTableRowInputEngineTest extends AbstractDbTest {
+
+	private final EmfEnumTableRowInputEngine<EmfTestEnumRow> inputEngine;
+
+	public EmfEnumTableRowInputEngineTest() {
+
+		this.inputEngine = new EmfEnumTableRowInputEngine<>(EmfTestEnumRow.TABLE);
+	}
+
+	@Test
+	public void testWithEmptyPattern() {
+
+		assertEquals("[]", toDisplayStrings(inputEngine.findMatches("", 0)));
+		assertEquals("[Five]", toDisplayStrings(inputEngine.findMatches("", 1)));
+		assertEquals("[Five, Four, One]", toDisplayStrings(inputEngine.findMatches("", 3)));
+		assertEquals("[Five, Four, One, Three, Two]", toDisplayStrings(inputEngine.findMatches("", 5)));
+		assertEquals("[Five, Four, One, Three, Two]", toDisplayStrings(inputEngine.findMatches("", 10)));
+	}
+
+	@Test
+	public void testWithOneLetterPattern() {
+
+		assertEquals("[]", toDisplayStrings(inputEngine.findMatches("f", 0)));
+		assertEquals("[Five]", toDisplayStrings(inputEngine.findMatches("f", 1)));
+		assertEquals("[Five, Four]", toDisplayStrings(inputEngine.findMatches("f", 2)));
+		assertEquals("[Five, Four]", toDisplayStrings(inputEngine.findMatches("f", 3)));
+	}
+
+	@Test
+	public void testWithTwoLetterPattern() {
+
+		assertEquals("[]", toDisplayStrings(inputEngine.findMatches("fi", 0)));
+		assertEquals("[Five]", toDisplayStrings(inputEngine.findMatches("fi", 1)));
+		assertEquals("[Five]", toDisplayStrings(inputEngine.findMatches("fi", 2)));
+	}
+
+	@Test
+	public void testWithOneLetterPatternAndTranslation() {
+
+		CurrentLocale.set(new Locale().setLanguage(LanguageEnum.GERMAN));
```

Comment:
It's actually great to now have test coverage for this!

Reply:
Indeed :)